data/changelog/ppxlib/2024-07-25-ppxlib-0.33.0.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Ppxlib 0.33.0
+tags: [ppxlib, platform]
+changelog: |
+  - Fix a bug where `Code_path.main_module_name` would not properly remove
+  extensions from the filename and therefore return an invalid module name.
+  (#512, @NathanReb)
+
+  - Add `-unused-type-warnings` flag to the driver to allow users to disable
+  only the generation of warning 34 silencing structure items when using
+  `[@@deriving ...]` on type declarations. (#511, @mbarbin, @NathanReb)
+
+  - Make `-unused-code-warnings` flag to the driver also controls the generation
+  of warning 34 silencing structure items when using `[@@deriving ...]` on type
+  declarations. (#510, @mbarbin, @NathanReb)
+
+  - Driver: Add `-unused-code-warnings=force` command-line flag argument. (#490, @mbarbin)
+
+  - New functions `Ast_builder.{e,p}list_tail` that take an extra tail
+  expression/pattern argument parameter compared to `Ast_builder.{e,p}list`, so
+  they can build ASTs like `a :: b :: c` instead of only `[ a; b ]`.
+  (#498, #502, @v-gb, @NathanReb)
+
+  - Fix `Longident.parse` so it also handles indexing operators such as
+  `.!()`, `.%(;..)<-`, or `Vec.(.%())` (#494, @octachron)
+
+  - Add a `special_function'` variant which directly takes a `Longident.t`
+  argument to avoid the issue that `Longident.t` cover distinct syntaxic classes
+  which cannot be easily parsed by a common parser (#496, @octachron).
+
+  - Keep location ranges consistent when migrating `Pexp_function` nodes from 5.2+
+  to older versions (#504, @jchavarri)
+
+  - Fix `-locations-check` behaviour so it is no longer required to pass `-check` and
+  can enable location checks. (#506, @NathanReb)
+---
+
+The Ppxlib dev team is happy to announce the release of `ppxlib.0.33.0`.
+
+The main feature of this release is a series of improvement to flags controlling
+the silencing of unused value/module/type warnings.
+The `ppxlib` driver generates warning silencing items to prevent `[@@deriving ...]`
+generated code to trigger unused code warnings. Three warnings are disabled that
+way:
+- Warning 32: unused value
+- Warning 60: unused module
+- Warning 34: unused type
+The first two are disabled for values and modules generated by the deriver while
+the third is disabled for the types in the type declaration to which the
+`[@@deriving ...]` attribute is attached.
+This feature was added a long time ago to avoid manually disabling those
+warnings when working with derivers that generate a set of values and modules
+only to use a subset of those. Alternatively, the unused type warning silencing
+was added to allow defining an alias type only to be consumed by a deriver (e.g.,
+``type error = [`Not_found | `Invalid_arg] [@@deriving to_string]``).
+We since then believe that we should not disable warnings lightly and this
+behaviour makes it difficult to find and remove deadcode. The right approach in
```

Comment:
```suggestion
We since then believe that we should not disable warnings lightly, as this
behaviour makes it difficult to find and remove deadcode. The right approach in
```